[kineto] add SOFT_ASSERT when logging metdata

chaekit opened the pull request:
Summary: having a valid `kineto_activity_` before logging metadata is a crucial invariant worthy of asserts

Test Plan:
## Test with D44362040

Verify that we get SOFT_ASSERT logs before and after the diff

## Log
```
W0329 11:29:34.269824 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270107 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270385 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270653 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270941 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.271199 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.271476 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.271724 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272003 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272280 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272553 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272822 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.273092 718148 profiler_kineto.cpp:122] Warning:  (function operator())
```

Reviewed By: aaronenyeshi

Differential Revision: D44513152

aaronenyeshi reviewed (approved):
LGTM, reviewed internally

chaekit committed:
[kineto] add SOFT_ASSERT when logging metdata (#98442)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/98442

having a valid `kineto_activity_` before logging metadata is a crucial invariant worthy of asserts

Test Plan:
## Test with D44362040

Verify that we get SOFT_ASSERT logs before and after the diff

## Log
```
W0329 11:29:34.269824 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270107 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270385 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270653 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.270941 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.271199 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.271476 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.271724 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272003 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272280 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272553 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.272822 718148 profiler_kineto.cpp:122] Warning:  (function operator())
W0329 11:29:34.273092 718148 profiler_kineto.cpp:122] Warning:  (function operator())
```

Reviewed By: aaronenyeshi

Differential Revision: D44513152

fbshipit-source-id: 60eb53248290993f3128ad8dc89c55bda58e2691
